sqrl-planner/sqrl-planner-local/src/main/java/com/datasqrl/sql/FlinkSqlGenerator.java

```diff
@@ -0,0 +1,453 @@
+      TableDescriptorSourceFactory sourceFactory = (TableDescriptorSourceFactory) connectorFactory;
+      FlinkSourceFactoryContext factoryContext = new FlinkSourceFactoryContext(null, name, tableConfig.serialize(),
+          formatFactory, UUID.randomUUID());
+      TableDescriptor descriptor = getTableDescriptor(sourceFactory, factoryContext, table.getSchema());
+
+      return toCreateTable(name, descriptor,(RelDataType) table.getRelDataType(),
+          (Optional<SqlNode>)table.getWatermarkExpression(),
+          (Optional<SqlNode>)table.getWatermarkColumn(),
+          sourceFactory.getSourceTimeMetaData(), true,
+          table.getSchema().getPrimaryKey()
+          );
+    } else if (connectorFactory instanceof SinkFactory) {
+      SinkFactory sinkFactory = (SinkFactory) connectorFactory;
+      Object o = sinkFactory.create(
+          new FlinkSinkFactoryContext(name, tableConfig, formatFactory));
+      if (o instanceof DataStream) {
+        throw new RuntimeException("Not supported");
+      } else if (o instanceof TableDescriptor.Builder) {
+        TableDescriptor.Builder builder = (TableDescriptor.Builder) o;
+        TableDescriptor descriptor = builder.schema(toSchema(table.getSchema()))
+            .build();
+
+        return toCreateTable(name, descriptor, (RelDataType) table.getRelDataType(),
+            (Optional<SqlNode>) table.getWatermarkExpression(),
+            (Optional<SqlNode>) table.getWatermarkColumn(), Optional.empty(),
+            false, table.getSchema().getPrimaryKey());
+      } else {
+        throw new RuntimeException("Unknown sink type");
+      }
+
+    } else {
+      throw new RuntimeException("Unknown factory type " + connectorFactory.getClass().getName());
+    }
+  }
+
+  private SqlCreateTable toCreateTable(String name, TableDescriptor descriptor, RelDataType relDataType,
+      Optional<SqlNode> watermarkExpression, Optional<SqlNode> watermarkColumn,
+      Optional<String> sourceTimeMetaData, boolean setMetadata, List<String> primaryKey) {
+    return new SqlCreateTable(SqlParserPos.ZERO,
+        identifier(name),
+        createColumns(descriptor.getSchema(), relDataType, watermarkExpression, watermarkColumn, sourceTimeMetaData, setMetadata), //columns
+        createConstraints(descriptor.getSchema(), primaryKey), //table constraints
+        createProperties(descriptor.getOptions()), //property list ?
+        createPartitionKeys(descriptor.getPartitionKeys()), //partition key list
+        createWatermark(descriptor.getSchema(),watermarkColumn, watermarkExpression, setMetadata),
+        createComment(descriptor.getComment()),
+        true,
+        false
+    );
+  }
+
+  private SqlWatermark createWatermark(Optional<Schema> schema,
+      Optional<SqlNode> watermarkColumn, Optional<SqlNode> watermarkExpression,
+      boolean setMetadata) {
+    if (!setMetadata) {
+      return null;
+    }
+    final String watermarkName;
```

Comment:
We'll refactor this for 0.5 because we need to use the watermark configured in the table.json.